src/query/service/src/interpreters/interpreter_delete.rs

```diff
@@ -234,10 +235,15 @@ impl Interpreter for DeleteInterpreter {
             )
             .await?
         {
+            let mut segments = Vec::with_capacity(snapshot.segments.len());
```

Comment:
If the number of segments is too small, there will be a problem of  the data skew.